packages/components/src/alignment-matrix-control/README.md

```diff
@@ -1,76 +1,86 @@
+<!-- This file is generated automatically and cannot be edited directly. -->
+
 # AlignmentMatrixControl
 
-AlignmentMatrixControl components enable adjustments to horizontal and vertical alignments for UI.
 
-## Usage
+<p class="callout callout-info">See the <a href="https://wordpress.github.io/gutenberg/?path=/docs/components-alignmentmatrixcontrol--docs">WordPress Storybook</a> for more detailed, interactive documentation.</p>
+
+AlignmentMatrixControl components enable adjustments to horizontal and vertical alignments for UI.
 
 ```jsx
-import { useState } from 'react';
 import { AlignmentMatrixControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
 
 const Example = () => {
 	const [ alignment, setAlignment ] = useState( 'center center' );
 
 	return (
 		<AlignmentMatrixControl
 			value={ alignment }
-			onChange={ ( newAlignment ) => setAlignment( newAlignment ) }
+			onChange={ setAlignment }
 		/>
 	);
 };
 ```
-
 ## Props
 
-The component accepts the following props:
+### `defaultValue`
 
-### className
+If provided, sets the default alignment value.
 
-The class that will be added to the classes of the underlying `grid` widget.
--   Type: `string`
--   Required: No
+ - Type: `"center" | "top left" | "top center" | "top right" | "center left" | "center center" | "center right" | "bottom left" | "bottom center" | "bottom right"`
+ - Required: No
+ - Default: `'center center'`
 
-### id
+### `label`
 
-Unique ID for the component.
+Accessible label. If provided, sets the `aria-label` attribute of the
+underlying `grid` widget.
 
--  Type: `string`
--  Required: No
+ - Type: `string`
+ - Required: No
+ - Default: `'Alignment Matrix Control'`
 
-### label
+### `onChange`
 
-Accessible label. If provided, sets the `aria-label` attribute of the underlying `grid` widget.
+A function that receives the updated alignment value.
 
--   Type: `string`
--   Required: No
--   Default: `Alignment Matrix Control`
+ - Type: `(newValue: AlignmentMatrixControlValue) => void`
```

Comment:
Is there a way we can be more consistent with type expansion? In this case, for example, `AlignmentMatrixControlValue` is expanded for the `value` prop but not for the `onChange` prop 🤔

Reply:
It's consistent in the sense that the output pretty much matches Storybook.

But no, the docgen doesn't give me details for `AlignmentMatrixControlValue` in the `onChange` prop data, so we can't really do anything there.

Reply:
Got it. So, nothing to do for the time being. If we ever want to tweak/improve this aspect, we'll have to look at a different types extractor for both Storybook and this docgen